app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -119,6 +128,29 @@ public boolean install(Resource r, ResourceLocation location, Reference ref,
         return false;
     }
 
+    private void storeReleasedTime(AndroidCommCarePlatform platform, Reference ref) {
+        long releasedOnTime = -1;
+        if (ref instanceof JavaHttpReference) {
+            Headers responseHeaders = ((JavaHttpReference)ref).getResponseHeaders();
+            if (responseHeaders != null) {
+                String releasedOnStr = responseHeaders.get(HEADER_APP_RELEASED_ON);
+                try {
+                    if (releasedOnStr != null) {
+                        releasedOnTime =  new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'").parse(releasedOnStr).getTime();
+                    }
+                } catch (ParseException e) {
+                    e.printStackTrace();
```

Comment:
I'm not sure we actually want to eat this error silently. A parse error here means that HQ is setting the header, but in the wrong format, which means that the phones will all just silently fail to properly use the feature correctly, but without anyone's knowledge.